pkg/refactor/logstore/logstore.go

```diff
@@ -0,0 +1,67 @@
+package logstore
+
+import (
+	"context"
+	"time"
+
+	"github.com/direktiv/direktiv/pkg/util"
+	"github.com/google/uuid"
+)
+
+// LogStore manages storing and querying LogEntries.
+type LogStore interface {
+	// appends a log-entry to the logs of the corresponding RecipientID, RecipientType pair.
+	Append(recipientID uuid.UUID, recipientType RecipientType, l LogEntry) error
+	// returns a list of log-entries using the provided LogQuery.
+	Get(ctx context.Context, ql LogQuery) ([]*LogEntry, error)
+}
+
+// Represents an individual log entry for activity.
+type LogEntry struct {
+	// the timestamp of the log-entry.
+	T     time.Time
+	Msg   string
+	Level Level
+	// Tags contains metadata of the log-entry.
+	Tags util.JSONB
+}
+
+// LogQuery generates a query statement to receive Log-Entries.
+type LogQuery interface {
```

Comment:
It's not clear how the users of LogStore would use this interface as parameter to query logs!

Can you please write an example?